flocker/ca/_ca.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Low-level logic for a certificate authority.
+
+Uses ECDSA + SHA256? Or if that turns out to be too difficult RSA 4096-bit
++ SHA 256.
```

Comment:
Ooooohhh boy.  I remember this game.  Wait wait is the answer     

```
ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:
DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS
```

Oops no that's the cheat code for the sequel.

SHA-256 still seems to be a good thing.  Definitely not SHA-1 (which you didn't suggest, just sayin').  Unfortunately I think you'll find the APIs necessary for ECDSA missing from pyOpenSSL.  There are probably other Python libraries for doing it, though (whether any are any good not, I dunno; I wonder if cryptography offers a good API for this yet).